src/ORM/OrmCache.php

```diff
@@ -90,17 +92,31 @@ public function clear($name = null)
     /**
      * Helper method to get the schema collection.
      *
-     * @param string $connection Connection name to get the schema for
-     * @return false|\Cake\Database\Schema\Collection|\Cake\Database\Schema\CachedCollection
+     * @param string|\Cake\Datasource\ConnectionInterface $connection Connection name to get the schema for or a connection instance
+     * @return \Cake\Database\Schema\Collection|\Cake\Database\Schema\CachedCollection
      */
     public function getSchema($connection)
     {
-        /* @var \Cake\Database\Connection $source */
-        $source = ConnectionManager::get($connection);
+        if (is_string($connection)) {
```

Comment:
I don't mind if the method only took connection instance. That would allow typehinting the argument and removing checks. Since this class is not going to be used often in userland code I don't think having to do `new SchemaCache(ConnectionManager::get('default'))` would be too much of a inconvenience.

We could also make the constructor argument optional and use "default" connection if no argument is passed, as that's what's gonna be used in most cases anyway.